src/server/cluster/cluster_config.cc

```diff
@@ -120,7 +126,7 @@ optional<SlotSet> ClusterConfig::SetConfig(const vector<ClusterShard>& new_confi
           shard.master.id == my_id_ || any_of(shard.replicas.begin(), shard.replicas.end(),
                                               [&](const Node& node) { return node.id == my_id_; });
       for (SlotId i = slot_range.start; i <= slot_range.end; ++i) {
-        if (slots_[i].owned_by_me && !owned_by_me) {
+        if ((slots_[i].owned_by_me || is_first_config) && !owned_by_me) {
```

Comment:
I think it will be cleaner, and probably a better API, to not modify this function (`SetConfig()`), but instead perform this check at the call site (i.e. in `ClusterFamily::DflyClusterConfig()`).

Reply:
As discussed, it's easier to keep as is, so I withdraw this comment :)